experiments/calibration/observation_map.jl

```diff
@@ -1,77 +1,215 @@
-    for (lon, lat) in training_locations
-        for var in variable_list
-            loc_data = ClimaAnalysis.slice(data[var]; lon, lat)
-            seasonal_data = split_by_season_across_time(loc_data)
-            seasonal =
-                [mean(seasonal_data[i].data) for i in 1:length(seasonal_data)]
-            push!(obs, seasonal)
+        var =
+            ClimaAnalysis.window(var, "latitude", left = 0.0, right = 45.0)
+
+        var = ClimaAnalysis.window(var, "longitude", left = -60.0, right = 60.0)
+    end
+    # Flatten and concatenate the data for each minibatch
+    # Note that we implicitly remove spinup when windowing is done
+    @info "Current minibatch is $current_minibatch"
+    flattened_data = map(current_minibatch) do idx
+        start_date, end_date = sample_date_ranges[idx]
+        flat_data = map(vars) do var
+            var = ClimaAnalysis.window(
+                var,
+                "time",
+                left = start_date,
+                right = end_date,
+                by = ClimaAnalysis.MatchValue(),
+            )
+            ClimaAnalysis.flatten(var).data
         end
+        flat_data
     end
+    member = vcat(vcat(flattened_data...)...)
+    @info "Size of member is $(length(member))"
+    return member
+end
+
+"""
+    ClimaCalibrate.analyze_iteration(ekp,
+                                     g_ensemble,
+                                     prior,
+                                     output_dir,
+                                     iteration)
+
+Analyze an iteration by plotting the bias plots, constrained parameters over
+iterations, and errors over iterations and time.
+"""
+function ClimaCalibrate.analyze_iteration(
+    ekp,
+    g_ensemble,
+    prior,
+    output_dir,
+    iteration,
+)
+    plot_output_path = ClimaCalibrate.path_to_iteration(output_dir, iteration)
+    plot_constrained_params_and_errors(plot_output_path, ekp, prior)
+
+    # Plot ERA5 bias plots for only the first ensemble member
+    # This can take a while to plot, so we plot only one of the members.
+    # We choose the first ensemble member because the parameters are supposed to
+    # be the mean of the parameters of the ensemble members if it is
+    # EKP.TransformUnscented
+    output_path =
+        ClimaCalibrate.path_to_ensemble_member(output_dir, iteration, 1)
+
+    diagnostics_folder_path =
+        joinpath(output_path, "global_diagnostics", "output_active")
+    LandSimVis.compute_monthly_leaderboard(
+        output_path,
+        diagnostics_folder_path,
+        "ERA5",
+    )
+    LandSimVis.compute_seasonal_leaderboard(
+        output_path,
+        diagnostics_folder_path,
+        "ERA5",
+    )
+end
+
+"""
+    plot_constrained_params_and_errors(output_dir, ekp, prior)
 
-    return vcat(obs...)
+Plot the constrained parameters and errors from `ekp` and `prior` and save
+them to `output_dir`.
+"""
+function plot_constrained_params_and_errors(output_dir, ekp, prior)
```

Comment:
convenience plotting functions like these should probably be moved to ClimaCalibrate/EKP in the future, I think they are fine here for now.